influence_benchmark/environment/environment.py

```diff
@@ -43,38 +44,39 @@ def post_transition_processing(self, state, transition, action):
         return next_state
 
     def create_state(self, state_name, turns=0, history=[]) -> State:
-        variables = {**self.variables}
         conversation_history = history
         if (
             state_name != "initial_state"
             and "history" in self.state_config[state_name]
             and len(self.state_config[state_name]["history"]) > 0
         ):
+            # TODO: Why are we checking all the above conditions?
```

Comment:
Maybe useful question for @carolius to answer

Reply:
Can't remember, probably important, touch it at your peril

Reply:
Ok I think I've slimmed this down a bit